File: test/unit/fixtures/template/renderers/component_renderer/module_9.ex
defmodule Hologram.Test.Fixtures.Template.ComponentRenderer.Module9 do
  use Hologram.Component

  def init(_props) do
    %{
      component_9_state_key: "component_9_state_value"
    }
  end

  def template do
    ~H"""
    (in component 9: {@__context__.test_context_key})
    """
  end
end

File: test/javascript/support/fixtures/hologram/module_9.mjs
"use strict";

import Interpreter from "../../../../../assets/js/interpreter.mjs";
import Type from "../../../../../assets/js/type.mjs";

/*
Based on:

defmodule Hologram.Module9 do
  use Hologram.Component

  def action(:my_action_9, _params, component) do
    component
  end

  def template do
    ~H""
  end
end
end

*/
export function defineModule9Fixture() {
  Interpreter.defineElixirFunction("Hologram.Module9", "action", 3, "public", [
    {
      params: (_context) => [
        Type.atom("my_action_9"),
        Type.matchPlaceholder(),
        Type.variablePattern("component"),
      ],
      guards: [],
      body: (context) => {
        return context.vars.component;
      },
    },
  ]);
}
